lib/hanami/application.rb

```diff
@@ -48,6 +49,10 @@ def self.extended(klass)
         end
       end
 
+      def application
+        self
+      end
```

Comment:
After I'm done with this PR, one of the next things I want to look at is making it so the application is just a specialised form of slice. This is a tiny step towards this, because it means we can allow `SliceConfigurable` and it users just to expect a `slice` to be given, not "a `slice` or an `application`", which is what led me to use the ill-fitting term "provider" to represent this union in previous iterations of this code.